package.json

```diff
@@ -48,7 +48,8 @@
   "dependencies": {
     "async": "^2.6.1",
     "glob": "^7.0.3",
-    "lodash": "^4.8.2"
+    "lodash": "^4.8.2",
+    "resolve-pkg": "^1.0.0"
```

Comment:
If we merge `resolve-pkg`, I think I get a free coffee for having 10 different dependencies that all implement the resolve algorithm.

Off the top of my head, [quibble relies on substack/browserify `resolve`](https://github.com/testdouble/quibble/blob/master/package.json#L22) whereas [testdouble uses the built-in `require.resolve`](https://github.com/testdouble/testdouble.js/blob/1bbf695953912d9c2925bd8c570c9ff0c3328dbf/src/replace/module/require-actual.js#L22). Is there any chance this change could rely on one of those two instead?

Reply:
The issue with `require.resolve` is that if your module does not contain a "main" property it will not resolve. So when I was testing locally with our internal module it would not resolve because of this. We could force a "blank" main file but that seems a little weird (and out of scope of this). Also the result of `require.resolve` is the entry path (package root + main). This means we would have to do some processing to "strip out" these extra paths. The same issue exists in `resolve`.

resolve-pkg's tag line is: 
> Resolve the path of a package regardless of it having an entry point

I agree that bringing in new packages should be avoided at all costs. However, it seems like the 3 options are to either bring this in, write a bunch of custom logic to do the resolve, or force a convention on the modules. My option is that the first options seems like the best fit. I am open to any counter points?

Reply:
Welp, ok then.